### ISSUE TITLE
Make it possible to run multiple che-server in parallel

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#7662]
### Description

This is about starting multiple wsmaster servers (behind a reverse proxy) to serve the same group of workspaces for the purpose of:
- [blue-green deployment](https://martinfowler.com/bliki/BlueGreenDeployment.html) to minimize downtime during new versions rollout
- load balancing

**Comment by skabashnyuk** [comment]
Can we consider this issue is related to wsmaster servers of the same version?

**Comment by l0rd** [comment]
@skabashnyuk this mechanism should work even for wsmaster with different versions except, of course, if there are breaking changes in the API.

What I mean is that the goal is to realease to production many times a day. To achieve it we need blue-green deployment. So for the deployments that do not introduce breaking changes in the API (99% of the deployment) we should be able run 2 wsmaster in parallel and for deployments that do introduce breaking changes (the remaining 1%) we should use different techniques like canary release or pre-release migration jobs (but this epic is not about these!).

**Comment by garagatyi** [comment]
I like the idea!
FYI this should relate to multi-user packaging only since single-user packaging uses DB embedded into Che instance.

**Comment by garagatyi** [comment]
Here are issues I see as needed to solve this epic:
- get rid of the volume mounted to /data path of Che master container (not sure whether it contains anything useful or is present only because it was done in that way in Docker-based deployment)
- implement runtime picking up #5919
- rework auth subsystem to be stateless #7785
- make sure data transactions into DB allow us to run 2 instances of Che
- make workspace activity checking feature stateless 
- feature toggle technics for testing blue-green deployments (is this already done for rh-che)

**Comment by l0rd** [comment]
Reopening. To be revisited once https://github.com/eclipse/che/issues/15425 is completed

**Comment by skabashnyuk** [comment]
@l0rd can you explain the goal of this task?  
Is it suppose to see multiple DevWorksapce controllers in parallel or what?

**Comment by l0rd** [comment]
@skabashnyuk no I am still thinking about the che server and the devworkspace controller proxy.

**Comment by skabashnyuk** [comment]
so then it's about making che-server stateless? Right?

**Comment by l0rd** [comment]
@skabashnyuk yes

**Comment by ibuziuk** [comment]
I belive we need to close this issue as outdated since the workspace engine is now DevWorkspace operator - https://newsroom.eclipse.org/eclipse-newsletter/2022/may/eclipse-che-gets-new-dev-environments-engine

**Comment by ibuziuk** [comment]
currently, che deployment is using rolling update 
```
 strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 25%
      maxSurge: 25%
```
the plan is to retest the flow with workspace startup during rollout update once we decommission db and che-server would become stateless - https://github.com/eclipse/che/issues/21374

**Comment by ibuziuk** [comment]
@l0rd I would propose closing this issue in favour of https://github.com/eclipse/che/issues/22067
Basically, it is possible to run multiple che-server in parallel (demoed on the last sprint review). What we are missing is options for configuring replicas for each operand on the CR level

**Comment by l0rd** [comment]
@ibuziuk makes sense 👍
